server/parser/src/main/antlr4/org/eclipse/lsp/cobol/core/CobolLexer.g4

```diff
@@ -918,8 +918,10 @@ COMMASEPARATOR : ', '  {enableCobolSpecialSeparators}? -> channel(HIDDEN);
 // PICTURECLAUSE mode
 // -------------------------------------------------------
 mode PICTURECLAUSE;
-FINALCHARSTRING: CHARSTRING+ ->popMode;
+FINALCHARSTRING: (CHARSTRING+ | SINGLE_UTF_CHARSTRING) ->popMode;
```

Comment:
I am not sure this approach is right or should we deal with it [here](https://github.com/eclipse-che4z/che-che4z-lsp-for-cobol/blob/f8be5ea66b8e4661ff1827886bc02869232036c1/server/parser/src/main/antlr4/org/eclipse/lsp/cobol/core/CobolParser.g4#L662)

Reply:
This obviously, saves some validation processor classes and BYTE-LENGTH lexeme collision with other rules like cobolWord

Reply:
This approach will make it harder to extract the number if it is ever needed.
I'd suggest trying to ban the single `U` from FINALCHARSTRING, then redefined `charString` as `FINALCHARSTRING | U` and modify the `dataPictureClause` rule as `U (BYTE_LENGTH PICTUREIS? INTEGERLITERAL)? | charString`. Not sure what the purpose of `+` is in that rule.